### ISSUE TITLE
TypeError: Cannot read property 'indexOf' of undefined

### ISSUE THREAD

**Issue by daveykropf** [Siilwyn/css-declaration-sorter#61]
```
Module build failed (from ../node_modules/@kaliber/build/webpack-loaders/css-loader.js):
TypeError: Cannot read property 'indexOf' of undefined
    at /Users/dkropf/Documents/Sites/projectname/node_modules/css-declaration-sorter/src/index.js:47:47
    at /Users/dkropf/Documents/Sites/projectname/node_modules/postcss/lib/container.js:144:26
    at Root.each (/Users/dkropf/Documents/Sites/projectname/node_modules/postcss/lib/container.js:110:22)
    at Root.walk (/Users/dkropf/Documents/Sites/projectname/node_modules/postcss/lib/container.js:143:21)
    at processCss (/Users/dkropf/Documents/Sites/projectname/node_modules/css-declaration-sorter/src/index.js:41:7)
    at /Users/dkropf/Documents/Sites/projectname/node_modules/css-declaration-sorter/src/index.js:22:14
    at initializePlugin (/Users/dkropf/Documents/Sites/projectname/node_modules/cssnano/dist/index.js:35:51)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

A fix would be:
```javascript
const isNewlineNode = ~node.raws.before.indexOf('\n');
```
to
```javascript
const isNewlineNode = node.raws.before && ~node.raws.before.indexOf('\n');
```

For now it is not possible for us to upgrade to postcss 7. So this means I cannot create a pull request for the current master branch (version 4.*).

My question; Is it possible to create a `3.0.2` version with this fix? Or is possible to create a 3.x branch so I can create a pull request?

**Comment by Siilwyn** [comment]
Thank you for your descriptive issue submission. I've never come across a 'CSS node' with no `raws`, could you give the input you use? Will need to write a failing test first, thanks!

Regarding backporting this fix to version `3.0.2`, will do! :)

**Comment by Siilwyn** [comment]
Hi there @daveykropf did you find time to give an example?

**Comment by EECOLOR** [comment]
@daveykropf things have changed, we are on the latest version, is this still an issue?

**Comment by EECOLOR** [comment]
@Siilwyn I dove in a bit deeper and found this line in `postcss`: https://github.com/postcss/postcss/blob/7.0.23/lib/node.es6#L38

```js
class Node {
  constructor (defaults = { }) {
    this.raws = { }
    ...
```

This means that postcss states that `raws` is not optional. From which you can conclude that plugins / libraries can assume `raws` exists. Libraries and plugins should however not assume `before` or `after` to exist. This argument becomes even stronger with: https://github.com/postcss/postcss/blob/7.0.23/lib/node.es6#L388

```js
  cleanRaws (keepBetween) {
    delete this.raws.before
    delete this.raws.after
    ...
```

So I would suggest performing the check on the existence `before` and `after` before usage.

**Comment by Siilwyn** [comment]
@EECOLOR thanks a ton for the complete investigation digging into this issue! I would more than welcome you to be a contributor if you'd like, feel free to accept or decline.

@daveykropf would you still like this fix to be backported to `3.0.2`? I'm working on a big update with a breaking change so thinking of including this in that version bump. :thought_balloon:
